src/utils/provider.ts

```diff
@@ -82,12 +82,22 @@ export function addAuthorize<
           code_verifier: codeVerifier,
           code
         }
-
-        const headers = {
+        /* eslint no-var: */
+        var headers = {
           Accept: 'application/json',
           'Content-Type': 'application/json'
         }
 
+        if (
+          tokenEndpoint.toLowerCase() ===
+          'https://accounts.spotify.com/api/token'
```

Comment:
This is leaking Spotify specifics out of the Spotify provider and into the general provider code. We need to separate those concerns. How can this be uncoupled such that the main provider class can be configured to do the same thing, with Spotify using that configuration option?

Reply:
While developing a PR for this I found out, that passing the `client-secret` in the payload of the post-request is also accepted. 
[Official Spotify documentation](https://developer.spotify.com/documentation/general/guides/authorization/code-flow/) states, that the `client-secret` and the `client-id` needs to be passed in the Authorization header but the alternative works as well.

However the [OAuth 2.0 spec](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1) states 

> Including the client credentials in the request-body using the two parameters is NOT RECOMMENDED and SHOULD be limited to clients unable to directly utilize the HTTP Basic authentication scheme (or other password-based HTTP authentication schemes). 

But in the current implementation the `client-secret` is passed either way in the payload of the post-request.

What do you think @bmulholland @Intevel ?

Reply:
@michelmueller Yeah, that's a good point. Will be tricky to implement without breaking people's setup. It's also more general than Spotify, so I think it's worth splitting into a separate issue (though may end up being a dependency of this PR?)

Reply:
I suggest, that the spotify section in the general provider code in this PR gets removed. So the spotify provider works and does not violate the OAuth 2.0 specification. Honestly I've never contributed to a project yet. So hopefully it is not rude to ask you to do that @Intevel ? :-)

@bmulholland  Agree. But we could add another configuration option to the `Oauth2SchemeOptions` which defaults to the current behavior. When you change the behavior in the config, the `client-secret` gets moved from the body to the HTTP Authorization header. This should be no breaking change.

Reply:
As it looks this should not be a problem, I'll comment it out for now. What solution do we have for this problem instead? 
And of course it is no problem to ask @michelmueller :-)

Reply:
I've drafted a potentiable solution in this PR (https://github.com/nuxt-community/auth-module/pull/1656)